src/Razor/src/Microsoft.VisualStudio.LanguageServer.ContainedLanguage/DefaultLSPRequestInvoker.cs

```diff
@@ -79,26 +77,43 @@ internal class DefaultLSPRequestInvoker : LSPRequestInvoker
                 throw new ArgumentException("message", nameof(method));
             }
 
-            if (capabilitiesFilter == null)
+            if (capabilitiesFilter is null)
             {
                 capabilitiesFilter = _fallbackCapabilitiesFilterResolver.Resolve(method);
             }
 
             var serializedParams = JToken.FromObject(parameters);
 
-            var (_, resultToken) = await _languageServiceBroker.RequestAsync(
-                new[] { contentType },
+            var (languageClient, resultToken) = await _languageServiceBroker.RequestAsync(
+                Array.Empty<string>(),
                 capabilitiesFilter,
                 languageServerName,
                 method,
                 serializedParams,
                 cancellationToken);
 
-            var result = resultToken != null ? resultToken.ToObject<TOut>(_serializer) : default;
+            var result = resultToken != null ? new ReinvokeResponse<TOut>(languageClient, resultToken.ToObject<TOut>(_serializer)) : default;
             return result;
         }
 
-        private async Task<IEnumerable<TOut>> RequestMultipleServerCoreAsync<TIn, TOut>(string method, string contentType, Func<JToken, bool> capabilitiesFilter, TIn parameters, CancellationToken cancellationToken)
+        public Task<TOut> ReinvokeRequestOnServerWithClientAsync<TIn, TOut>(string method, string languageServerName, TIn parameters, CancellationToken cancellationToken)
```

Comment:
>public [](http://example.com/codeflow?start=8&length=6)

Do this and the other WithClient method need to be public still?

Reply:
You're right, I was trying to leave them as a convenience, but nobody accessed this as `DefaultLSPRequestInvoker`, it's all `LSPRequestInvoker` (as it should be) so there's no point.